### ISSUE TITLE
AutoCast option not working for arrays

### ISSUE THREAD

**Issue by gianlucadonato** [fabioricali/valify#11]
Error:
`"location.coordinates.0" expects "number" type but receives: ["123", "321"]"`

Code:
```javascript
const data = new Valify({
  foo: 'string?',
  location: new Valify({
    coordinates: ['number']
  }, { autoCast: true })
})

const result = data({
   location: { coordinates: ["123","321"] }
})
```

I suggest to fix this in "castToPrimitiveType" function (valify.js line 568) 
from:

```javascript
static castToPrimitiveType(field, data) {
  if (data.hasOwnProperty(field))
    data[field] = detectType(data[field]);
}
```

To:

```javascript
static castToPrimitiveType(field, data) {
  if (data.hasOwnProperty(field)) {
    if (Array.isArray(data[field]) {
     data[field] = data[field].map(val => detectType(val));
    } else {
      data[field] = detectType(data[field]);
    }   
  }
}
```

**Comment by fabioricali** [comment]
Hi Gianluca,

thanks for your post. You can create a PR?

**Comment by fabioricali** [comment]
Fixed in 4.4.2
